memory/memhotplug.py

```diff
@@ -0,0 +1,250 @@
+       1. hotunplug one by one in a loop for all
+       2. Toggle memory blocks by making off/on in a loop
+       3. hot unplug % of memory for different ratios
+       4. dlpar memory hotplug using  drmgr
+       5. shared resource : dlpar in CMO mode
+       6. try hotplug each different numa node memblocks
+       7. run stress memory in background
+
+    :avocado: tags=memory,power,privileged
+    '''
+
+    def setUp(self):
+
+        if not memory.check_hotplug():
+            self.cancel("UnSupported : memory hotplug not enabled\n")
+
+        if os.geteuid() != 0:
+            exit("You need to have root privileges to run this script."
+                 "\nPlease try again, using 'sudo'. Exiting.")
+        sm = SoftwareManager()
+        deps = ['stress']
+        for package in deps:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.cancel(' %s is needed for the test to be run' % package)
+        self.tests = self.params.get('test', default='all')
+        self.iteration = int(self.params.get('iteration', default='1'))
+        self.stresstime = int(self.params.get('stresstime', default='5'))
+        self.blocks_hotpluggable = get_hotpluggable_blocks(
+            '%s/memory*' % mem_path)
+
+    @staticmethod
+    def hotunplug_all(blocks):
+        for block in blocks:
+            if memory._check_memory_state(block):
+                offline(block)
+
+    @staticmethod
+    def hotplug_all(blocks):
+        for block in blocks:
+            if not memory._check_memory_state(block):
+                online(block)
+
+    @staticmethod
+    def __clear_dmesg():
+        process.run("dmesg -c", sudo=True)
+
+    @staticmethod
+    def __error_check():
+        ERROR = None
+        logs = (process.system_output(
+            "dmesg -T --level=alert,crit,err,warn", ignore_status=True, shell=True, sudo=True)).split()
+        for log in logs:
+            for error in errorlog:
+                if error in log:
+                    ERROR = 'Test: resulted %s in syslogs"' % error
+        return ERROR
+
+    @staticmethod
+    def __is_auto_online():
+        with open('%s/auto_online_blocks' % mem_path, 'r') as auto_file:
+            if auto_file.read() == 'online\n':
+                return True
+            return False
+
+    @staticmethod
+    def run_stress(stresstime):
+        mem_free = memory.meminfo.MemFree.mb / 2
+        cpu_count = int(multiprocessing.cpu_count()) / 2
+        process.run("stress --cpu %s --io 4 --vm 2 --vm-bytes %sM --timeout %ss" %
+                    (cpu_count, mem_free, stresstime), ignore_status=True, sudo=True, shell=True)
+
+    def test(self):
+        if os.path.exists("%s/auto_online_blocks" % mem_path):
+            if not self.__is_auto_online():
+                self.hotplug_all(self.blocks_hotpluggable)
+        if 'all' in self.tests:
+            tests = ['hotplug_loop',
```

Comment:
please consider test  as yaml input it make every option as a single test for avocado run

Reply:
sure

Reply:
@abdhaleegit what about this review  comment please reply

Reply:
I have made the changes